plugins/Live/API.php

```diff
@@ -147,6 +148,10 @@ public function getLastVisitsDetails($idSite, $period = false, $date = false, $s
         }
         Piwik::checkUserHasViewAccess($idSite);
 
+        if (!Live::isVisitorLogEnabled($idSite) && (Request::isCurrentApiRequestTheRootApiRequest() || Request::getRootApiRequestMethod() === 'Live.getLastVisitsDetails')) {
```

Comment:
would this also for bulk requests? Might need to specifically only allow the method to be called when `API.getSuggestedValuesForSegment` is called as the root api request method. We don't really use that in bulk so it should work I suppose

Reply:
Changed that. The method `API.getSuggestedValuesForSegment` now only works when called directly and fails when called in a bulk request (if visits log is disabled)